release: release 2.0-rc version.

LiteSun opened the pull request:
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

This part will track all issues related to 2.0 RC3
- [x] #546 
- [x] #593 
- [x] #608 
- [x] https://github.com/apache/apisix-dashboard/issues/615
- [x] #617 
- [x] https://github.com/apache/apisix-dashboard/issues/621

LiteSun committed:
Update CHANGELOG.md

LiteSun committed:
Update CHANGELOG.zh-CN.md

LiteSun committed:
Update CHANGELOG.md

LiteSun committed:
Update CHANGELOG.zh-CN.md

membphis commented on CHANGELOG.md:
```diff
@@ -18,10 +18,26 @@
+# Core
+
+- Refactor front-end with admin-api.
+- Manage-api removes dependency on mysql.
```
`mysql` -> `MySQL`

juzhiyuan replied:
manager-api or Manger-API, I would prefer the first one.

membphis commented on CHANGELOG.md:
```diff
@@ -18,10 +18,26 @@
+
+# Core
+
+- Refactor front-end with admin-api.
```
`admin-api` -> `Admin-API`

juzhiyuan commented on CHANGELOG.md:
```diff
@@ -18,10 +18,26 @@
 
+# 2.0.0
+
+This is release candidate.
```
`a` release

juzhiyuan commented on CHANGELOG.md:
```diff
@@ -18,10 +18,26 @@
+
+- Refactor front-end with admin-api.
+- Manage-api removes dependency on mysql.
+- Support Plugin orchestration.
```
Plugin should be lowercase.

LiteSun committed:
Update CHANGELOG.md

LiteSun committed:
feat: update deploy docs

LiteSun committed:
feat: Merge branch 'v2.0' of github.com:apache/apisix-dashboard into v2.0

LiteSun committed:
Merge branch 'master' into v2.0

nic-chen committed:
feat: separate build and run to 2 scripts (#600)

* feat: separate build and run

* doc: update doc about build and run

* doc: update doc about build and run

* Update schema-sync.sh

* Update develop.md

* Update develop.zh-CN.md

Co-authored-by: 琚致远 <juzhiyuan@apache.org>

juzhiyuan committed:
feat: remove unused dist folder

juzhiyuan reviewed (approved):
LGTM

jiayx committed:
doc: add etcd api version tips (#604)

* doc: add etcd version tips(#602)

* Update README.md

* Update README.zh-CN.md

Co-authored-by: 琚致远 <juzhiyuan@apache.org>

juzhiyuan committed:
Update deploy.md

juzhiyuan committed:
Update deploy.zh-CN.md

juzhiyuan committed:
feat(docs): update deploy docs

juzhiyuan committed:
feat(doc): added tip in deploy

juzhiyuan committed:
feat(doc): update run part in deploy

juzhiyuan committed:
feat(doc): added FAQ about how to redeploy

nic-chen committed:
fix: 2.0 release issues (#606)

* fix: remove json schema generation  from ci

* fix: use json schema generated from APISIX v2.0

juzhiyuan reviewed (approved):
LGTM

juzhiyuan committed:
fix: ant-design/ant-design/issues/27396

nic-chen committed:
fix: using relative path to read conf (#617)

* fix: using  relative path

* fix path

* fix path

juzhiyuan reviewed (changes requested):
Just waiting for RC3

nic-chen committed:
feat: support custom server host, port and DAG lib path (#625)

* feat: support custom server port and dag lib path

* test: custom host

* test: add test case

* fix bug

* test

* test

* test

* feat support config etcd endpoints

* Update conf.json

* Update conf.go

* Update test-api.yml

* Update deploy.md

* Update deploy.zh-CN.md

* Update deploy.md

* Update conf.json

Co-authored-by: 琚致远 <juzhiyuan@apache.org>

juzhiyuan committed:
feat(doc): update config file

juzhiyuan committed:
feat(doc): update Note

juzhiyuan committed:
feat(docs): update NOTE

juzhiyuan committed:
feat(doc): update ETCD endpoints

juzhiyuan committed:
Update deploy.md

moonming commented on api/build-tools/schema-sync.sh:
```diff
@@ -0,0 +1,35 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+pwd=`pwd`
+
+version="master"
+if [[ -n $1 ]]; then
+version=$1
+fi 
+
+rm -rf ./api/build-tools/apisix/
+wget https://github.com/apache/apisix/archive/$version.zip
+
+unzip $version.zip
+mkdir -p ./api/build-tools/apisix/
+mv ./apisix-$version/apisix/* ./api/build-tools/apisix/
+rm -rf ./apisix-$version
+cd ./api/build-tools/ && lua schema-sync.lua > ${pwd}/api/conf/schema.json
+
+echo "sync success:" 
+echo "${pwd}/api/conf/schema.json"
```
We should delete all the logic in this file, and users don't need to care about these details at all. We generate these files and put them in the code repo. @nic-chen

juzhiyuan replied:
Ping @nic-chen

nic-chen replied:
sure, I will remove them, and put them in a repo temporarily, in case someone needs to use them, until the new json schema sync is implemented

moonming commented on api/build.sh:
```diff
@@ -0,0 +1,31 @@
+pwd=`pwd`
+
+# get dag-to-lua lib
+if [[ ! -f "dag-to-lua-1.1/lib/dag-to-lua.lua" ]]; then
+    wget https://github.com/api7/dag-to-lua/archive/v1.1.tar.gz
+    tar -zxvf v1.1.tar.gz
+fi
```
can we move this repo dashboard?

nic-chen replied:
sorry, do you mean mv `dag-to-lua`'s code into dashboard repo ?

moonming commented on api/conf/conf.go:
```diff
@@ -42,16 +41,50 @@ const (
+	ENV           string
+	basePath      string
+	Schema        gjson.Result
+	DagLibPath    = "/go/manager-api/dag-to-lua/"
```
I don't think root dir is good

juzhiyuan replied:
Yep, and developers couldn't find those directories actually 🤔

![image](https://user-images.githubusercontent.com/2106987/97891590-ef63d180-1d69-11eb-930b-0e4583ff0123.png)

nic-chen replied:
it's a history issue. working on it.

moonming commented on api/conf/conf.go:
```diff
@@ -61,23 +94,23 @@ func setEnvironment() {
 
 func configurationPath() string {
-	if ENV == EnvLOCAL {
+	if confPath := os.Getenv("APISIX_CONF_PATH"); confPath != "" {
```
how about move to configure file too?

nic-chen replied:
@moonming  working on it.

moonming commented on api/conf/conf.json:
```diff
@@ -2,6 +2,14 @@
+    },
+    "dag-lib-path": "",
+    "etcd": {
+      "endpoints": "127.0.0.1:2379"
```
how about etcd cluster?

juzhiyuan replied:
Just added the description in deploy.md

moonming commented on api/conf/schema.json:
```diff
@@ -1 +1 @@
-{"main":{"upstream_hash_header_schema":{"type":"string","pattern":"^[a-zA-Z0-9-_]+$"},"stream_route":{"type":"object","properties":{"plugins":{"type":"object"},"id":{"anyOf":[{"type":"string","minLength":1,"maxLength":64,"pattern":"^[a-zA-Z0-9-_]+$"},{"type":"integer","minimum":1}]},"remote_addr":{"type":"string","anyOf":[{"title":"IPv4","type":"string","pattern":"^[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3 ...
+{"main":{"plugin_disable_schema":{"disable":{"type":"boolean"}},"route":{"properties":{"create_time":{"type":"integer"},"desc":{"type":"string","maxLength":256},"vars":{"type":"array","items":{"description":"Nginx builtin variable name and value","items":{"maxItems":3,"minItems":2,"anyOf":[{"type":"string"},{"type":"number"}]},"type":"array"}},"uris":{"type":"array","items":{"description":"HTTP u ...
```
What is the purpose of this file? Is it automatically generated?

nic-chen replied:
sorry for reply late.I think we had discuss it in mail list.

moonming commented on docs/deploy.zh-CN.md:
```diff
@@ -88,4 +76,83 @@ $ yarn build
+
+## 启动
+
+1. 根据您的部署环境，检查并修改 `api/conf/conf.json` 中的配置。
```
This directory is too deep, can it be shortened later?

juzhiyuan replied:
yep, I'm refactoring the frontend and backend.

juzhiyuan committed:
feat: refactor folders (#629)

* feat: refactor api & frontend

* feat: remove some actions

* feat: added - v2.0

* feat: added ignored files for checker

* feat: trigger ci

* feat: trigger CI

* feat: added conf.json

* fix: build dashboard

* feat: remove demo temp

juzhiyuan committed:
feat: update build.sh

juzhiyuan committed:
chore: improve issue template and vscode (#660)

nic-chen committed:
test: add e2e test for field hosts in `route` api (#612)

* feat: ETCD cluster and APISIX cluster for CI

* feat: add test cases

* feat: add upstream service in docker compose

* fix: comment

* test: add test cases

* fix: code format

* test: add ci

* fix ci

* fix ci

* fix ci

* fix: remove consumer test

* test: sleep for sync

* test: e2e as an independent subproject

* fix CI error

* test: run docker

* fix: remove json schema generate script in docker build

* fix: check host and hosts config together

* fix ci

* test: add test cases for `host` in route

* remove useless code

* fix: using relative path to read conf (#617)

* fix: using  relative path

* fix path

* fix path

* fix conf path

* fix ci error

* fix etcd ip

* fix: code format

* run backend e2e test ci on v2 branch

* fix: code format

* fix: code format

* fix: CI error

* test: remove deploy CI again

* remove useless codes

* fix: go fmt

* test: don't use `go fmt`

* fix: code format

* fix: var name

* fix lint

* fix CI error

* debug

* fix: docker container name

* fix CI error

* fix CI error

* test: add more test cases

* fix CI error

* chore: remove useless code

* fix: go fmt

* fix: refactor test code

* fix: check body

* fix small issue

* fix: update docker compose

* fix CI

* test

* test

* test: build image first

* test: revert subnet ip in docker compose

* fix by review

* fix: docker compose

* test: add test cases

* fix: remove useless code

* fix: code format

* test: code format

LiteSun committed:
feat: added e2e test for Login page (#619)

* feat: added Front-end e2e test YAML file

* feat: added login e2e test

* feat: update Login.e2e.js

* Update Login.e2e.js

* feat: added e2e readme

* feat: added licence

* feat: added  start-server-and-test package

* feat: update login test case

* Update frontend-e2e-test.yml

* feat: added logout test case

* Update frontend-e2e-test.yml

* feat: added login failed with empty input

* feat: update CI

* feat: update text

* feat: added public.js

* feat: change logout timeout

* feat: Added e2e test documentation link to development.md

* Update develop.md

* Update develop.zh-CN.md

* Update README.md

Co-authored-by: 琚致远 <juzhiyuan@apache.org>

juzhiyuan commented:
Just to remind, there still have some comments that waiting for @nic-chen's reply.

nic-chen committed:
CI: collect golang unit test code coverage (#654)

LiteSun committed:
fix: promethus incorrect value when update route (#666)

juzhiyuan committed:
feat: deploy with Docker (#657)

* feat: added Docker

* feat: added License header

* feat: added line in docs

* Update test-docker.yml

* feat(doc): update docs

* feat: added extra line

* feat: added Go Proxy in Dockerfile

* feat: update Dockerfile

* chore: update docs

* fix: copy correct files

* feat: improve Dockerfile

* Revert "feat: improve Dockerfile"

This reverts commit c68a4c4e479147e65efae4611ba162bd2c465928.

* fix run fail

* feat: update docs

* fix: compatible with Golang conf

* Squashed commit of the following:

commit 94450bf6ad52cdcac8d527c7e7fbf9e8fa1c4ab3
Author: litesun <7sunmiao@gmail.com>
Date:   Tue Nov 3 10:19:04 2020 +0800

    fix: promethus incorrect value when update route (#666)

commit de8bdbf6ff867a05dffbd05151b1ba07603f46d2
Author: nic-chen <33000667+nic-chen@users.noreply.github.com>
Date:   Tue Nov 3 07:50:29 2020 +0800

    CI: collect golang unit test code coverage (#654)

commit 722c0fdeddbc34723f730e36a4cff004889f7093
Author: litesun <7sunmiao@gmail.com>
Date:   Tue Nov 3 00:03:06 2020 +0800

    feat: added e2e test for Login page (#619)

    * feat: added Front-end e2e test YAML file

    * feat: added login e2e test

    * feat: update Login.e2e.js

    * Update Login.e2e.js

    * feat: added e2e readme

    * feat: added licence

    * feat: added  start-server-and-test package

    * feat: update login test case

    * Update frontend-e2e-test.yml

    * feat: added logout test case

    * Update frontend-e2e-test.yml

    * feat: added login failed with empty input

    * feat: update CI

    * feat: update text

    * feat: added public.js

    * feat: change logout timeout

    * feat: Added e2e test documentation link to development.md

    * Update develop.md

    * Update develop.zh-CN.md

    * Update README.md

    Co-authored-by: 琚致远 <juzhiyuan@apache.org>

commit 915ce8300b8138eb986737e6df1201fff63d3b57
Author: nic-chen <33000667+nic-chen@users.noreply.github.com>
Date:   Mon Nov 2 21:13:34 2020 +0800

    test: add e2e test for field hosts in `route` api (#612)

    * feat: ETCD cluster and APISIX cluster for CI

    * feat: add test cases

    * feat: add upstream service in docker compose

    * fix: comment

    * test: add test cases

    * fix: code format

    * test: add ci

    * fix ci

    * fix ci

    * fix ci

    * fix: remove consumer test

    * test: sleep for sync

    * test: e2e as an independent subproject

    * fix CI error

    * test: run docker

    * fix: remove json schema generate script in docker build

    * fix: check host and hosts config together

    * fix ci

    * test: add test cases for `host` in route

    * remove useless code

    * fix: using relative path to read conf (#617)

    * fix: using  relative path

    * fix path

    * fix path

    * fix conf path

    * fix ci error

    * fix etcd ip

    * fix: code format

    * run backend e2e test ci on v2 branch

    * fix: code format

    * fix: code format

    * fix: CI error

    * test: remove deploy CI again

    * remove useless codes

    * fix: go fmt

    * test: don't use `go fmt`

    * fix: code format

    * fix: var name

    * fix lint

    * fix CI error

    * debug

    * fix: docker container name

    * fix CI error

    * fix CI error

    * test: add more test cases

    * fix CI error

    * chore: remove useless code

    * fix: go fmt

    * fix: refactor test code

    * fix: check body

    * fix small issue

    * fix: update docker compose

    * fix CI

    * test

    * test

    * test: build image first

    * test: revert subnet ip in docker compose

    * fix by review

    * fix: docker compose

    * test: add test cases

    * fix: remove useless code

    * fix: code format

    * test: code format

commit 94d024544df0caff6aef69f6ca9792954a285304
Author: 琚致远 <juzhiyuan@apache.org>
Date:   Mon Nov 2 13:17:42 2020 +0800

    chore: improve issue template and vscode (#660)

* Revert "Squashed commit of the following:"

This reverts commit 93d38eef07058f5c65ac1eb83aac9d04ec12d24d.

* feat: remove output from ignore files

Co-authored-by: nic-chen <johz@163.com>

juzhiyuan committed:
feat: use web instead of frontend (#674)

* feat: use web instead of frontend

* feat: rename frontend to web

* feat: remove all frontend to web

liuxiran committed:
feat: support get plugin schema based on schema_type (#651)

* feat: support get plugin schema based on schema_type

* fix: ci errors

* fix: run error casued by api-breaker.lua

* fix: get schema_type from query and add some test

* fix: update schema.json file

* fix: update validate to support schematype

* fix: properties:{} validate failed

* fix: some code errors refer to the review

nic-chen committed:
test: add linter for manager api (#655)

* test: add go lint

* fix lint

* fix by review

* fix errors

* test: run lint on v2.0

* fix: text format

* fix: os check

* fix typo

* fix: fmt --> log

* fix log

* fix: unnecessary nil check around range for lint

* fix: trigger lint for push to branch v2.0

juzhiyuan committed:
feat: update plugin to 1.0.10

juzhiyuan commented:
need more attention for those comments, if they are resolved, kindly resolve then.

cc @nic-chen

moonming commented:
Since there is no feedback, should we close this pr?

juzhiyuan commented:
No, just ping @nic-chen

LiteSun committed:
fix: enable HTTPS setting unsuccessful in Route (#692)

* fix: wrong stepHeader after edit MatchingRules

* feat: disable plugin orchestration when select forcehttps

* feat: update Route transform

LiteSun committed:
feat: cache Front-end e2e CI node_modules (#696)

nic-chen committed:
chore: refactor `conf` of `manager api` (#693)

* feat: refactor conf

* fix default listen port

* fix build and run scripts

* fix: docs

* chore: remove useless file

* fix docker for test

* fix CI

* fix CI

* fix ci

* fix: `-c` conf dir  -->  `-p` work dir

* fix go test error

* fix conf

* fix: revert changes

* fix: remove useless comment

* fix: remove useless comment

* doc: add comment for config

* doc: todo

* fix: config format

* fix: if secret use default value, should generate a random string to replace it.

* fix comment style

* fix: change web dir in docker file

* doc: update comments

nic-chen committed:
fix:bug that dirty data exists after updating route and wrong mod for prod env (#704)

* fix: route bug, dirty data exists after updating

* fix mod for env

* fix mod for env

* fix error log

* fix error log

nic-chen committed:
chore: refactor error log for `manager api` (#689)

* chore: refactor log

* fix: custom log by conf

* feat: add error log

* fix default config

* fix CI fail

* fix: should not save log to file by default

* test: add test case

* test: add test case

* fix CI fail

* fix error

* fix CI

* fix error

* fix according to reviews

* test: more test cases

* fix error

* chore: use `/dev/stdout` as default log file path

* fix typo

* fix docker for logs dir

* fix CI fail

* fix: delete useless files

* fix: change file name

nic-chen committed:
bugfix: dashboard 2.0 failed to fetch ssl certificate not found  (#719)

* fix: set ssl status, since it's default 0

* add test cases

* test: add test cases

* fix: update test cases

* fix: test case fail

* test: remove hosts in hosts setting

* fix: according review

* fix: according review

* chore: add comment

juzhiyuan committed:
fix: update order (#744)

juzhiyuan committed:
feat(Consumer): use username instead of id (#742)

* feat(Consumer): use username instead of id

* feat: remove duplicated var

membphis committed:
chore: exit if any error and specify the download file name when download by `wget`. (#751)

fix #646

nic-chen committed:
fix: an error will occur if `pass_host` is set to `node` when creating upstream (#750)

* fix: bug #749

* fix: bug #749

* test: add test cases

* fix: CI

* fix: CI fail

nic-chen committed:
test: add e2e test cases for upstream (#738)

juzhiyuan committed:
fix: react warnings (#747)

juzhiyuan commented:
cc @nic-chen not sure why this E2E will fail.

nic-chen committed:
fix: update json schema (#754)

nic-chen committed:
chore: move the Dockerfile to `test` folder, because it was used for testing (#753)

nic-chen commented:
> cc @nic-chen not sure why this E2E will fail.

checking

juzhiyuan committed:
feat: convert uri to uris (#740)

juzhiyuan committed:
feat: deploy with docker (#701)

* feat: added deploy with docker CD

* feat: docker deploy test

* fix: CI

* fix: path of `config.yaml`

* fix: CI fail

* docs: update doc for docker deploy

* fix: typo

* fix: add EOF && trigger  CD

Co-authored-by: nic-chen <johz@163.com>
Co-authored-by: kv <gxthrj@163.com>

juzhiyuan committed:
fix: invalid values from the manager-api (#736)

* feat: return None when timestamp is invalid

* chore: added TODO

gxthrj committed:
docs: add makefile && modify  develop and deploy  docs (#729)

* docs: add makefile && modify  develop and deploy  docs

* docs: Make the makefile clearer

* docs: modify frontend to web

* doc: two blankline between commands

* doc: remove blanklien EOF

* docs: remove make run/stop in makefile

* docs: make readme readable

* docs: declear in readme

* docs: make readme readable

* docs: remove dependencies in readme

* docs: check english desc

* docs: style adjust

* docs: sync makefile desc

* docs: remove blank

* docs: style adjust

* docs: remove ENV=local in deploy

* fix: modify lint ci

* docs: unify go-lint

* docs: check all text

* docs: makefile Aligned

* docs: update style

* docs: add markdown code style

* docs: fix  syntax

* docs: remove Self-referencing

* docs: user guide

* docs: use header

* docs: style fix

* docs: add desc

* docs: modify 'pack' to 'package'

* docs: unify manager-api

* docs: mv Install to Installation

* docs: node to Node.js

* docs: source codes to Source Codes

* docs: Source Codes

* fix: mkdir -p ./output/logs in makefile build

nic-chen committed:
feat: support specifying APISIX path to generate json schema (#765)

* feat: support specifying APISIX path to generate json schema

* doc: update doc about schema sync

* fix: doc

* fix: errors

* fix: error according to test

* test

* no message

* test

* fix: remove debug

idbeta committed:
test: add consumer e2e test (#735)

* add consumer e2e test
add public method "PartialBody"

* test: add end with EOL

* add test data plane to case 2
delete some useless code

* modify code style

gxthrj committed:
fix: makefile build error (#767)

nic-chen committed:
fix: used dashboard add consumer of jwt, would have an error when get the jwt token (#768)

LiteSun committed:
feat: update changelog (#771)

* Update CHANGELOG.md

* Update CHANGELOG.zh-CN.md

* docs: update change log

Co-authored-by: nic-chen <johz@163.com>

nic-chen committed:
fix: consumer schema for auth plugin (#770)

* fix: consumer schema for auth plugin

* fix: update by jsonschema

* fix: json schema

nic-chen committed:
fix: doc (#772)

nic-chen committed:
fix: doc (#774)

juzhiyuan committed:
feat: remove CD for PR

starsz committed:
fix: closed WatchResponse channel when cancel function is called (#779) (#795)

nic-chen committed:
ci: fix CI naming (#799)

nic-chen committed:
fix: make cli test more compatible (#798)

nic-chen committed:
test: fix e2e test unstable (#800)

liuxiran committed:
fix: double scroll bar in plugin page (#801)

juzhiyuan committed:
feat(docs): improve README & Deploy (#785)

* feat(doc): update README & deploy

* feat(docs): added FAQ

* fix: linkx

* feat(docs): update deploy with docker

* feat(docs): update FAQ

* feat: update README

* feat: update README

* feta: update README

* fix: link

* feat: update deploy with docker

* feat(docs): added tip for some users

* feat(doc): improve deploy with docker

* feat(docs): improve deploy with docker

* feat(doc): update run with docker

* feat(doc): improve deploy

* feat(doc): added more info

* feat(doc): added more info

* feat(doc): remove extra info

* feat: added more detailed

* fix: CI fail according `api/conf/conf.yaml ` is  changed.

* fix: revert changed

* feat(docs): improve English version

* feat(docs): use frontend instead, just like backend

* feat(docs): remove extra statements

* feat(docs): update typo

* feat(docs): remove startup with message

* feat: update CI's name

Co-authored-by: nic-chen <johz@163.com>

LiteSun committed:
feat: skip puppeteer chromium download when build (#808)

LiteSun committed:
feat(Makefile): update release-src (#816)

nic-chen committed:
ci: fix CI fail (#818)

* ci: fix CI fail

* ci: fix CI fail

* ci: fix CI fail

EnableAsync committed:
test: add e2e test for config route with service_id or upstream_id (#810)

* test: add e2e test for config route with service_id or upstream_id

* test: fix route test

* test: fix route test

* test: fix route test

* test: fix route test format

* test: fix route test format

* test: fix test format

* test: add test on data plane

* test: fix test conflict

starsz committed:
feat: install signal handler for graceful shutdown (#737) (#796)

juzhiyuan committed:
Revert "test: add consumer e2e test (#735)" (#829)

This reverts commit c140f41acdc90b53db3b349f63a43503666a76fe.

ShiningRush committed:
feat: add a hanlder unit test for upstream and remove init

ShiningRush committed:
append license

ShiningRush committed:
revert unreviewed pr (#841)
